numcodecs/pickles.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+
+
+import numpy as np
+
+
+from numcodecs.abc import Codec
+from numcodecs.compat import ndarray_from_buffer, buffer_copy
+try:
+    import cPickle as pickle
+except ImportError:
+    import pickle
+
+
+class Pickle(Codec):
+    """Codec to encode data as as pickled bytes. Useful for encoding python
+    strings
+
+    Raises
+    ------
+    encoding a non-object dtyped ndarray will raise ValueError
+
+    Examples
+    --------
+    >>> import numcodecs as codecs
+    >>> import numpy as np
+    >>> x = np.array(['foo', 'bar', 'baz'], dtype='object')
+    >>> f = codecs.Pickle()
+    >>> f.decode(f.encode(x))
+    array(['foo', 'bar', 'baz'], dtype=object)
+
+    """  # flake8: noqa
+
+    codec_id = 'x-pickle'
+
+    def encode(self, buf):
+        if hasattr(buf, 'dtype') and buf.dtype != 'object':
+            raise ValueError("cannot encode non-object ndarrays, %s "
+                             "dtype was passed" % buf.dtype)
+        return pickle.dumps(buf, protocol=pickle.HIGHEST_PROTOCOL)
```

Comment:
Could you add ``protocol`` as a configuration parameter of the codec. This would mean adding as a keyword argument to ``__init__``, and also including in the output from ``get_config()``.